xml-conduit/Text/XML/Stream/Parse.hs

```diff
@@ -550,20 +567,27 @@ parseIdent =
     valid c  = not $ isXMLSpace c
 
 parseContent :: DecodeEntities
+             -> Bool -- preserve XML whitespace
              -> Bool -- break on double quote
              -> Bool -- break on single quote
              -> Parser Content
-parseContent de breakDouble breakSingle =
+parseContent de preserveWhiteSpace breakDouble breakSingle =
     parseEntity <|> parseText'
   where
     parseEntity = do
         char' '&'
-        t <- takeWhile1 (/= ';')
+        t <- takeWhile1 (/= ';') <?> "not a semicolon"
```

Comment:
`takeWhile` produces terrible error messages ("failed satisfy"), naming the parsers makes them more meaningful.